oasislmf/computation/generate/files.py

```diff
@@ -230,16 +235,24 @@ def run(self):
         gul_inputs_df = get_gul_input_items(
             location_df,
             keys_df,
-            output_dir=self._get_output_dir(),
             exposure_profile=location_profile,
             group_id_cols=group_id_cols,
-            hashed_group_id=self.hashed_group_id,
+            hashed_group_id=self.hashed_group_id
         )
         correlation_input_items = get_correlation_input_items(
             model_settings_path=self.model_settings_json,
             gul_inputs_df=gul_inputs_df
         )
 
+        correlations: bool = establish_correlations(model_settings_path=self.model_settings_json)
```

Comment:
I don't think you should have to read twice model_settings_path. get_correlation_input_items should return all the information that you need. So I would remove establish_correlations